### PR TITLE
add filters generically to meinberlin IdeaCollection

### PR DESCRIPTION
**Pull request by der-john**
this continues #2370:

as announced in #2347, IdeaCollection now gets the badges to be filtered from the backend; the `has-swot-labels` logic is replaced.

### CODE REVIEW

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Assignment.html [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-                <h2 class="badges-group-header">{{badgeGroups[groupPath].title}}</h2>
+                <h2 class="badges-group-header">{{badgeGroups[groupPath].title | translate}}</h2>
                 <label class="badge-toggle" data-ng-repeat="badgePath in badgePaths">
                     <input
                         type="checkbox"
                         name="badgePath"
                         class="badge-checkbox"
                         data-ng-model="checkboxes[badgePath]"/>
-                        {{badges[badgePath].title}}
+                        {{badges[badgePath].title | translate}}
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Assignment.html]: I don't think we should translate badge titles. Badges are user generated content and therefore we do not generally have translations for them.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 2]
```diff
@@ -14,10 +14,13 @@ import RIBuergerhaushaltProposalVersion from "../../../../Resources_/adhocracy_m
 import RIGeoProposalVersion from "../../../../Resources_/adhocracy_core/resources/proposal/IGeoProposalVersion";
 import RIKiezkasseProposalVersion from "../../../../Resources_/adhocracy_meinberlin/resources/kiezkassen/IProposalVersion";
 
+
```
Comment (xi) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: too many blank lines

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 3]
```diff
@@ -25,6 +25,43 @@ import * as SIWorkflow from "../../../../Resources_/adhocracy_core/sheets/workfl
 
 var pkgLocation = "/Meinberlin/IdeaCollection/Process";
 
+var bindFacets = (
```
Comment (xi) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: The name is misleading. Either rename this to `getFacets()` or wrap it in a `scope.$watch()`.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 4]
```diff
@@ -18,40 +19,68 @@ import * as SIImageReference from "../../../../Resources_/adhocracy_core/sheets/
...
+    adhHttp: AdhHttp.Service<any>,
+    $q: angular.IQService
+) => (
+    scope
+): void => {
+    var params = {
+        elements: "content",
+        depth: 4,
+        content_type: "adhocracy_core.resources.badge.IBadge"
```
Comment (xi) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: Sould be `SIBadge.nick`

**Comment by xi** [comment]
This adds generic badge filters to the proposal listing in meinberlin ideacollection. We could also think about making this a general feature of the adhListing.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/IdeaCollection.ts [hunk 5]
```diff
@@ -117,6 +119,14 @@ export var proposalDetailColumnDirective = (
...
             });
             adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
+            var params = {
+                elements: "content",
+                depth: 4,
+                content_type: SIBadge.nick
+            };
+            adhHttp.get(scope.processUrl, params).then((response) => {
+                scope.badgesExist = response.data[SIPool.nick].elements.length > 0;
+            });
```
Comment (xi) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/IdeaCollection.ts]: Better use  `response.data[SIPool.nick].count` and `elements: "omit"`.

Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/IdeaCollection.ts]: You're right. I'm on it.

**Comment by xi** [comment]
I did significant refactoring on `getFacets()`. Please review.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 6]
```diff
@@ -27,28 +27,23 @@ import * as SIWorkflow from "../../../../Resources_/adhocracy_core/sheets/workfl
...
 var createBadgeGroups = (scope, groupPaths, badges) => {
     scope.badgesByGroup = AdhBadge.collectBadgesByGroup(groupPaths, badges);
-    var facets = [];
-    _.forOwn(scope.badgeGroups, (group, groupPath) => {
-        var badgeGroup = createBadgeGroup(scope, group, groupPath);
-        facets.push(badgeGroup);
+    return _.map(scope.badgeGroups, (group, groupPath) => {
+        return createBadgeGroup(scope, group, groupPath);
     });
-    return facets;
```
Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: Nice!

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 7]
```diff
@@ -90,7 +90,9 @@ export var detailDirective = (
         },
         require: "^adhMovingColumn",
         link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
-            getFacets(adhHttp, $q)(scope);
+            getFacets(adhHttp, $q)(scope.path).then((facets) => {
+                scope.facets = facets;
+            });
```
Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: good idea.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 8]
```diff
@@ -28,7 +28,7 @@ var pkgLocation = "/Meinberlin/IdeaCollection/Process";
 
 var createBadgeFacets = (badgeGroups, groupPaths, badges) => {
     var badgesByGroup = AdhBadge.collectBadgesByGroup(groupPaths, badges);
-    return _.map(badgeGroups, (group) => {
+    return _.map(badgeGroups, (group : any) => {
```
Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: why is that useful?

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts [hunk 9]
```diff
@@ -47,7 +48,7 @@ var getFacets = (
     $q : angular.IQService
 ) => (
     path : string
-) : angular.IPromise<any[]> => {
+) : angular.IPromise<AdhListing.IFacetItem[]> => {
```
Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/Process/Process.ts]: nice.

**Comment by der-john** [comment]
looks good & works.

**Pull request merged**